src/main/scala/com/tsunderebug/scolor/otf/tables/OTFNAMETable.scala

```diff
@@ -17,17 +17,24 @@ case class OTFNAMETable(
 
   override def sections(b: ByteAllocator): Seq[Section] = {
     val off = b.allocate(this)
-    val dataOff = off.position + UInt(6) + tabeledRecords.map(_.length(b)).toArray.qsum
+    val tabeledRecordsSum = tabeledRecords.foldLeft(UInt(0)) {
+      case (accum, tabledRecord) => accum + tabledRecord.length(b)
+    }
+    val dataOff = off.position + UInt(6) + tabeledRecordsSum
     Seq(
       Section("format", OTFUInt16(UShort(0))),
       Section("count", OTFUInt16(UShort(records.length))),
-      Section("stringOffset", OTFOffset16(6 + tabeledRecords.map(_.length(b)).toArray.qsum.toInt)),
+      Section("stringOffset", OTFOffset16(6 + tabeledRecordsSum.toInt)),
       Section("nameRecords", OTFArray(tabeledRecords)),
       Section("data", strData)
     )
   }
 
-  override def length(b: ByteAllocator): UInt = UInt(6) + records.map(_(this).length(b)).toArray.qsum + strData.length(b)
+  override def length(b: ByteAllocator): UInt = {
+    records.foldLeft(UInt(6) + strData.length(b)) { // Start with 6 because ???
```

Comment:
Hey @TsundereBug what is the 6 here anyway?

Reply:
It's the length of the data in bytes before the record data.